File: Data Structures & Algorithms/two-integer-sum/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
    let map = new Map();

    for (let i = 0; i <= nums.length - 1; i++) {
        map.set(nums[i], i);
    }

    for (let i = 0; i <= nums.length - 1; i++) {
        let complement = target - nums[i];
        let position = map.get(complement);
        if(position != undefined && position != i) return [i, position];
        else continue;
    }

    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        ans = []
        hashMap = {}
        found = False
        for index,value in enumerate(numbers):
            if value in hashMap:
                temp = hashMap.get(value)
                temp.append(index)
                hashMap[value] = temp
            else:
                hashMap[value] = [index]

        for index, number in enumerate(numbers):
            if not found:
                remainder = target - number
                if remainder in hashMap:
                    arr = hashMap[remainder]
                    for j in arr:
                        if j != index:
                            found = True
                            ans.append(index + 1)
                            ans.append(j + 1)
                            break
            else:
                break
        return ans